witness/golang/internal/persistence/persistence.go

```diff
@@ -25,8 +25,10 @@ type LogStatePersistence interface {
 	// be read.
 	Logs() ([]string, error)
 
-	// ReadOps returns read-only operations for the given log ID. This
-	// method only makes sense for IDs returned by Logs().
+	// ReadOps returns read-only operations for the given log ID.
+	// TODO(al): This method MUST return a valid LogStateReadOps even if
```

Comment:
That TODO is to prompt discussion as a first step - I didn't want to put a "proposed fix" in here, but it's worth thinking about whether the current interface contract (i.e. "hey, give me a readops for log X" - "sure!" - "use readops" - "whoa there buddy, I don't know what you're talking about") make sense vs. e.g. changing the higher levels to interpret a `NotFound` error on `ReadOps` equivalently to how it currently looks for one on `GetLatest`

Reply:
My goal when extracting out the persistence layer was very diligently not changing the witness, because that's the essence of good refactoring. If we want to go back and apply force in the other direction then that's an option.

Reply:
Yup, to be clear I totally agree that was the right approach, it's been helpful in identifying these sorts of hidden contract-y bits which makes it easier to talk/reason about what's happening in there.